### ISSUE TITLE
Missing timeout option for page loading

### ISSUE THREAD

**Issue by WWIJP** [sakuli/sakuli#211]
**Version**
Sakuli 2.1.2

**Short overview**
Our webpage is loading very slow so we need some time to wait (not a sleep).
Currently there is no timeout option available, it is hard coded

**Detailed error description**
Our webpage is loading very slow so we need some time to wait (not a sleep).
Currently there is no timeout option available, it is hard coded.
So wee need some timeout settings which we can set in the JS script.
like: "wait for the page to load x illiseconds"

**Steps to reproduce error**
starting a test case

**Additional content**
> Please provide any (mandatory) additional data to reproduce the error (Code, Config, Dockerfiles etc.)

**Comment by tnobody** [comment]
Hello @WWIJP,

thank you for your contribution.

I'm not sure about your intention here. In general `await _navigateTo(...)` will wait until the HTML file is delivered to the browser. After that a lot of things  can happen on the page (loading images, loading and executing javascript, loading some contents asynchronously, etc). Sakuli is currently not able to wait for all this (but we have a concept in mind to handle such scenarios in the future, issued in https://github.com/sakuli/sakuli/issues/115). 

So it will happen that the first action (a click for example) after the page is opened is not possible within it's own timeout (5 seconds). 

So you mean that the timeout of an action should be configurable, correct?

e.g.

```javascript
// init tc
try {
  await _navigateTo('long-loading.page');
  await _click(_button('loaded after 6 seconds'), 7000)
  // this should be configurable? ------------------^
} catch(e) {
// handle error
}
``` 

So that a action will have a longer timeout only in certain situations?

Br

**Comment by WWIJP** [comment]
Hello

exactly, we missing the option to slowdown the operation to be able to see the page in VNC.
This is currently not possible withthe methods: await env.sleep() or _wait().
And the taken error-screenshot was too slow for the exact picture.
We see the picture in VNC but it is just a flickering of it which is too short to read it.

Thanks

**Comment by WWIJP** [comment]
Hello

The current problem is, that the page is not shown in VNC about a certificate problem in Chrome.
So we are not abel to analyse the problem in the container.

Best regards

**Comment by ZNicoZ** [comment]
I think this could be solved with #251

**Comment by tnobody** [comment]
The root cause of the described problem was not related to any timeout. The test never started (so there was no chance to see an actual browser window in vnc) because of an syntax error in `package.json` file.

We introduced global timeout configuration with #251 anyway.

Therefore I'll close this issue.
